NeoMathEngine/src/CPU/CpuMathEngineSmallMatricesMultiplyDesc.cpp

```diff
@@ -0,0 +1,169 @@
+/* Copyright © 2023 ABBYY
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--------------------------------------------------------------------------------------------------------------*/
+
+#include <common.h>
+#pragma hdrstop
+
+#include <CpuMathEngine.h>
+#include <MemoryHandleInternal.h>
+#include <CpuMathEngineSmallMatricesMultiplyDesc.h>
+
+#ifdef NEOML_USE_MKL
+#if FINE_PLATFORM( FINE_WINDOWS ) || FINE_PLATFORM( FINE_LINUX ) || FINE_PLATFORM( FINE_DARWIN )
+#include <mkl.h>
+#else
+#error Unknown platform
+#endif
+#endif // NEOML_USE_MKL
+
+#define NEOML_MAX(a,b)  ((a)>(b)?(a):(b))
+
+//-------------------------------------------------------------------------------------------------------------------------
+
+namespace NeoML {
+
+inline CSmallMatricesMultiplyDesc* CCpuMathEngine::InitSmallMatricesMultiplyDesc(
+	int firstHeight, int firstWidth, int secondWidth, int secondRowSize, int resultWidth,
+	bool resultAdd, bool trans1, bool trans2 ) const
+{
+	return new CCpuSmallMatricesMultiplyDesc(
+		firstHeight, firstWidth, secondWidth, secondRowSize, resultWidth,
+		resultAdd, trans1, trans2 );
+}
+
+inline bool CCpuMathEngine::SmallMatricesMultiply( const CSmallMatricesMultiplyDesc* desc,
+	const CConstFloatHandle& firstHandle, const CConstFloatHandle& secondHandle, const CFloatHandle& resultHandle ) const
+{
+#if FINE_BIT( FINE_32_BIT )
+	return false;
+#endif // FINE_32_BIT
+
+#ifndef NEOML_USE_MKL
+	ASSERT_EXPR( false );
```

Comment:
First: this line is strange of itself (why `assert` instead of `return false`).
Second: you don't need this public function in interface. You only need private `smallMatricesMultiply` only on CPU. Which seems logical because this "small matrices" case is treated only on CPU.